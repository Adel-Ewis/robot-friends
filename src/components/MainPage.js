import React, { Component } from 'react';
import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import './MainPage.css';
import Scroll from '../components/Scroll';
import ErrorBoundry from '../components/ErrorBoundry';
import Header from '../components/Header';

class MainPage extends Component {
  componentDidMount() {
    this.props.onRequestRobots();
  }

  filteredRobots = () => {
    return this.props.robots.filter((robot) => robot.name.toLowerCase().includes(this.props.searchField.toLowerCase()));
  };
  render() {
    const { searchField, onSearchChange, isPending } = this.props;
    return isPending ? (
      <h1 className='tc'>Loading</h1>
    ) : (
      <div className='tc'>
        <Header />
        <SearchBox searchField={searchField} searchChange={onSearchChange} />
        <Scroll>
          <ErrorBoundry>
            <CardList robots={this.filteredRobots()} />
          </ErrorBoundry>
        </Scroll>
      </div>
    );
  }
}

export default MainPage;
