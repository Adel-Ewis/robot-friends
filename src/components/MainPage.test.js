import { shallow } from 'enzyme';
import MainPage from './MainPage';
import React from 'react';

let wrapper;
beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [
      { name: 'adel', email: 'adel@fscsa.com', id: 1 },
      { name: 'ahmed', email: 'adel@fscsa.com', id: 2 }
    ],
    searchField: '',
    isPending: false
  };
  wrapper = shallow(<MainPage {...mockProps} />);
});

describe('Testing MainPage Component', () => {
  it('expects to render MainPage Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Filters the robots correctly', () => {
    const mockProps2 = {
      onRequestRobots: jest.fn(),
      robots: [
        { name: 'adel', email: 'adel@fscsa.com', id: 1 },
        { name: 'ahmed', email: 'adel@fscsa.com', id: 2 }
      ],
      searchField: 'a',
      isPending: false
    };
    const wrapper2 = shallow(<MainPage {...mockProps2} />);
    expect(wrapper2.instance().filteredRobots()).toEqual(mockProps2.robots);
    expect(wrapper2.instance().filteredRobots().length).toEqual(2);
  });

  it('Check isPending when true', () => {
    const mockProps3 = {
      onRequestRobots: jest.fn(),
      robots: [],
      searchField: 'a',
      isPending: true
    };
    const wrapper3 = shallow(<MainPage {...mockProps3} />);
    expect(wrapper3.html()).toEqual('<h1 class="tc">Loading</h1>');
  });
});
