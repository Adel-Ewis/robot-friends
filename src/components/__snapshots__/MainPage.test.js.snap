// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MainPage Component expects to render MainPage Component 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox
    searchField=""
  />
  <Scroll>
    <ErrorBoundry>
      <CardList
        robots={
          Array [
            Object {
              "email": "adel@fscsa.com",
              "id": 1,
              "name": "adel",
            },
            Object {
              "email": "adel@fscsa.com",
              "id": 2,
              "name": "ahmed",
            },
          ]
        }
      />
    </ErrorBoundry>
  </Scroll>
</div>
`;
